### PR TITLE
Update list to remove bogus strengthened comment

### LINKED ISSUE #5020: `<list>`: Remove an incorrect `/* strengthened */` comment from the move assignment operator
Recently, when investigating stuffs in the standard library affected by WG21-N4258, I found that this exception specification isn't really strengthened:

https://github.com/microsoft/STL/blob/926d458f82ae1711d4e92c0341f541a520ef6198/stl/inc/list#L923-L924

The standard requirement (in [[list.overview]/2](https://eel.is/c++draft/list.overview#2)) is saying `noexcept(allocator_traits<Allocator>::is_always_equal::value)`, which hasn't changed since the adoption of N4258.

Looking at the definition of `_Choose_pocma_v`, it can be inferred that `_Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators` and `allocator_traits<_Alnode>::is_always_equal::value` are actually equal.

https://github.com/microsoft/STL/blob/926d458f82ae1711d4e92c0341f541a520ef6198/stl/inc/xmemory#L735-L746

It's not very clear whether we're allowed use this value, see LWG-3267. But it _is_ clear that this is _not_ strengthening!

I believe we should remove the misleading `/* strengthened */` comment.

Moreover, the exception specification of `deque` and (unordered) associative containers are spelt differently and more close to the forms specified in the standard. So perhaps we should spell the one of `list` as `noexcept(_Alnode_traits::is_always_equal::value)`.

----
There's another `_Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators` in `<xtree>`. I guess it would be more consistent to write `_Alnode_traits::is_always_equal::value` instead.

https://github.com/microsoft/STL/blob/926d458f82ae1711d4e92c0341f541a520ef6198/stl/inc/xtree#L943-L944

----
This issue is intended for a new contributor (especially one new to GitHub) to get started with the simplest possible change.

Please feel free to submit a pull request if there isn't one already linked here - no need to ask for permission! 😸

You can (and should) link your pull request to this issue using [GitHub's close/fix/resolve syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
(in the PR description not the commit message)

### PR DESCRIPTION
**Pull request by 5AIPAVAN**
Fixes #5020.

### CODE REVIEW

**Comment by AlexGuteniev** [comment]
clang format validation failed. run clang-format (from IDE or standalone) and update

**Comment by StephanTLavavej** [comment]
Additionally:

* Please edit your PR title to describe (very briefly) what your change is doing - mentioning issue numbers there is not useful. PR titles become part of git history (e.g. see `git log` or the graphical `gitk`), and a long series of "solve issue #nnnn" is completely opaque. "Update list to remove bogus strengthened comment" would be perfectly descriptive, or you can phrase it another way.
* Please edit your PR description to use GitHub's [fix/close/resolve syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) so that your PR is properly linked to the issue that it resolves. "Fixes #nnnn", "Resolves #nnnn", or "Closes #nnnn" are all fine.

Thanks!

File: stl/inc/list [hunk 1]
```diff
@@ -921,7 +921,7 @@ public:
     }
 
     list& operator=(list&& _Right)
-        noexcept(_Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators) /* strengthened */ {
+        noexcept(_Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators) {
```
Comment (frederick-vs-ja) [inline stl/inc/list:924]: Now you need to put these into a single line.

Would you mind to additionally simplify the spellings of the exception specifications mentioned in the issue?

Comment (5AIPAVAN) [reply stl/inc/list:924]: Can i change    
_Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators   
to
_Alnode_traits::is_always_equal::value(suggested by chat gpt)

My final change looks like this :-
list& operator=(list&& _Right) noexcept(_Alnode_traits::is_always_equal::value) {

Comment (frederick-vs-ja) [reply stl/inc/list:924]: > Can i change
> _Choose_pocma_v<_Alnode> == _Pocma_values::_Equal_allocators
> to
> _Alnode_traits::is_always_equal::value(suggested by chat gpt)

Hmm, this is OK. I also suggested this in the original issue. Would you mind to change that in `<xtree>` together?
But as a general rule, don't ask Chatgpt for bugfix, as it often gives wrong answer.

**Comment by AlexGuteniev** [comment]
You still haven't ran clang-format.

----

If you are trying to do commit using web interface only then I want to tell that this is generally not a path forward. 
Whereas you may be able to finish this change this way, it doesn't let you get any further, when you have to test and validate your changes locally. I recommend you learning how to use a Git/Github client and following `README.md` steps to build this locally.

(I admit I have done some web commits on my own, but the purpose of `good first issue` is for a contributor to learn the workflow; using web interface defeats this purpose)

**Comment by 5AIPAVAN** [comment]
> You still haven't ran clang-format.
> 
> If you are trying to do commit using web interface only then I want to tell that this is generally not a path forward. Whereas you may be able to finish this change this way, it doesn't let you get any further, when you have to test and validate your changes locally. I recommend you learning how to use a Git/Github client and following `README.md` steps to build this locally.
> 
> (I admit I have done some web commits on my own, but the purpose of `good first issue` is for a contributor to learn the workflow; using web interface defeats this purpose)

sure sir, actually i tried with some online clang formatters(they have shown that code is formatted correctly) and some vs code extensions, but haven't looked in depth
will try now and update by end of the day
thanks for your suggesions :-)

**Comment by 5AIPAVAN** [comment]
runned it locally 
cformat

**Review by frederick-vs-ja — APPROVED** [review]
LGTM assuming PR message fixed.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! I went ahead and edited your PR description to use GitHub's fix/close/resolve syntax.

This PR will be part of our next merge batch, likely next week. I'll post comments as I prepare your PRs for merging.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for the fix, and congratulations on your first microsoft/STL commit! :tada: :cat2: :smile_cat:

This change is expected to ship in VS 2022 17.13 Preview 2.

**Pull request merged**